pkg/cinder/cronjob.go

```diff
@@ -0,0 +1,131 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cinder
+
+import (
+	cinderv1 "github.com/openstack-k8s-operators/cinder-operator/api/v1beta1"
+
+	"fmt"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+)
+
+// CronJob func
+func CronJob(
+	instance *cinderv1.Cinder,
+	labels map[string]string,
+	annotations map[string]string,
+) *batchv1.CronJob {
+	cinderUser := int64(cinderv1.CinderUserID)
+	cinderGroup := int64(cinderv1.CinderGroupID)
+	config0644AccessMode := int32(0644)
+
+	debugArg := ""
+	if instance.Spec.Debug.DBPurge {
+		debugArg = " --debug"
```

Comment:
?: I thought that the `debug` section elements was meant to pause the execution to let the human operator/developer jump into the container and debug things manually.  Are we changing its meaning now?

Reply:
Well, I did update the comment (see cinder_types.go:203). I know the original use case was to pause execution, but "debug" is a generic term that can be broadly applied. I don't think it would be fair for the first use case (pausing execution) to own the rights to the entire ```debug``` section.

Reply:
That also makes sense, but would getting the debug info in the cron job be really helpful in the field instead of being able to get into the container from the cron job itself and debuging things manually?

Reply:
I believe the intent is to get the db_purge command to generate log messages. If they need to actually debug what happens when running the command then you don't actually need to pause the container to do that. You can rsh into a running container and manually run the command there.

Reply:
I'm not entirely convinced there's no value in pausing the cron job, but I don't have a strong opinion, so this code can stay as it is.

Since we are changing the pattern of pausing the container I think we should follow @fmount's suggestion and rephrase the comments to mention what kind of debugging each flag achieves.